### PR TITLE
build: Fix issues post-firehose

### PR DESCRIPTION
**Pull request by leoyvens**
Fixes a couple of issues falling out from https://github.com/graphprotocol/graph-node/pull/2716. One is the build failing with incremental compilation, which is fixed by disabling that. The other is having rustfmt as a build-time dependency, which is failing the docker build, this is worked around by formatting the tonic generated file only in debug builds.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,6 @@ env:
   CARGO_TERM_COLOR: always
   RUST_BACKTRACE: full
   THEGRAPH_STORE_POSTGRES_DIESEL_URL: "postgresql://postgres:postgres@localhost:5432/graph_node_test"
-  GRAPH_MAX_SPEC_VERSION: "0.0.4"
```
Comment (leoyvens) [inline .github/workflows/ci.yml:13]: This is an unrelated cleanup of an unnecessary setting.

**Comment by otaviopace** [comment]
Do we need to do any change for `release` mode?

**Review by lutter — APPROVED** [review]

**Comment by leoyvens** [comment]
@otaviopace release mode has incremental compilation off by default.
